### ISSUE TITLE
Records and Unneeded Boxing

### ISSUE THREAD

**Issue by JordanMartinez** [JordanMartinez/purescript-jordans-reference#8]
From [this](https://functionalprogramming.slack.com/archives/C04NA444H/p1533137169000123)
Explain differences between `data`, `type`, and `newtype` when using records
> If it’s a single wrapper you should be using a newtype
> data with a record is not like Haskell records, and it will introduce unnecessary runtime boxing

**Comment by JordanMartinez** [comment]
Returning to this issue,
```purescript
-- ideal, but defining typeclass instances on it
-- requires defining it for all records
type TypeWithRecord0 = { anInt :: Int }

-- allows defining type class instances,
-- but performs unnecessary unboxing
data TypeWithRecord1 = T { anInt :: Int }

-- use if it's a single wrapper
newtype TypeWithRecord2 = TWR { anInt :: Int }
```

**Comment by JordanMartinez** [comment]
Closed by in 4991b94
